integration/admin/admin_integration_test.go

```diff
@@ -81,6 +73,93 @@ func TestAdminServer_Unimplemented(t *testing.T) {
 	}
 }
 
+func TestAdminServer_CreateTree(t *testing.T) {
+	client, closeFn, err := setupAdminServer()
+	if err != nil {
+		t.Fatalf("setupAdminServer() failed: %v", err)
+	}
+	defer closeFn()
+
+	invalidTree := *testonly.LogTree
+	invalidTree.TreeState = trillian.TreeState_HARD_DELETED
+
+	tests := []struct {
+		desc string
+		req  *trillian.CreateTreeRequest
+		// TODO(codingllama): Check correctness of returned codes.Code
+		wantErr bool
+	}{
+		{
+			desc: "validTree",
+			req: &trillian.CreateTreeRequest{
+				Tree: testonly.LogTree,
+			},
+		},
+		{
+			desc:    "nilTree",
+			req:     &trillian.CreateTreeRequest{},
+			wantErr: true,
+		},
+		{
+			desc: "invalidTree",
+			req: &trillian.CreateTreeRequest{
+				Tree: &invalidTree,
+			},
+			wantErr: true,
+		},
+	}
+
+	ctx := context.Background()
+	for _, test := range tests {
+		createdTree, err := client.CreateTree(ctx, test.req)
+		if hasErr := err != nil; hasErr != test.wantErr {
+			t.Errorf("%v: CreateTree() = (_, %v), wantErr = %v", test.desc, err, test.wantErr)
+			continue
+		} else if hasErr {
+			continue
+		}
+		storedTree, err := client.GetTree(ctx, &trillian.GetTreeRequest{TreeId: createdTree.TreeId})
+		if err != nil {
+			t.Errorf("%v: GetTree() = (_, %v), want = (_, nil)", test.desc, err)
+			continue
+		}
+		if !reflect.DeepEqual(storedTree, createdTree) {
+			t.Errorf("%v: storedTree doesn't match createdTree:\nstoredTree =  %v,\ncreatedTree = %v", test.desc, storedTree, createdTree)
+		}
+	}
+}
+
+func TestAdminServer_GetTree(t *testing.T) {
+	client, closeFn, err := setupAdminServer()
+	if err != nil {
+		t.Fatalf("setupAdminServer() failed: %v", err)
+	}
+	defer closeFn()
+
+	tests := []struct {
+		desc    string
+		treeID  int64
+		wantErr bool
+	}{
+		{
+			desc:   "notFound",
+			treeID: -1,
```

Comment:
<paranoia> Maybe add another test case for not-found tree ID 999 (i.e. a positive number, so any `int` / `uint` conversions take a different path)?